src/java.net.http/share/classes/java/net/http/HttpRequest.java

```diff
@@ -654,5 +656,37 @@ public static BodyPublisher ofByteArrays(Iterable<byte[]> iter) {
         public static BodyPublisher noBody() {
             return new RequestPublishers.EmptyPublisher();
         }
+
+        /**
+         * Creates a {@code BodyPublisher} that publishes a request
+         * body consisting of the aggregation of the bytes published
+         * by a sequence of publishers.
+         *
+         * <p> If the sequence is empty an {@linkplain #noBody() empty} publisher
+         * is returned. Otherwise each publisher is lazily subscribed to in turn,
+         * until all the body bytes are published, an error occurs, or the
+         * returned publisher's subscription is cancelled.
```

Comment:
We should probably say what the publishers `contentLength` will return.

Reply:
OK.